### PR TITLE
GCE attach tests

### PR DESCRIPTION
**Pull request by jsafrane**
Add basic tests for GCE attacher.

Looking at the code, it would deserve some refactoring as suggested in #25888, so mounting is not tested at all.

### CODE REVIEW

**Comment by jsafrane** [comment]
@kubernetes/sig-storage

**Comment by jsafrane** [comment]
@k8s-bot: test this issue: #26210

**Comment by jsafrane** [comment]
@k8s-bot test this issue: #26210

**Comment by jsafrane** [comment]
Rebased. However, with simplifed `Attach`/`Deatch` from 9dbe943491d69762068c0ac021e1964d9c14873e the test is pretty straightforward and almost useless.

File: pkg/volume/gce_pd/attacher_test.go [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+	spec := createVSpec(diskName, readOnly)
+
+	err := attacher.Attach(spec, instanceID)
+	if err == nil {
+		t.Errorf("Attach failure expected, got nil")
+	}
+}
+
+func TestAttach_Positive_DiskIsAttached(t *testing.T) {
+	// DiskIsAttached query fails, but the plugin still tries to attach the
```
Comment (pmorie) [inline pkg/volume/gce_pd/attacher_test.go]: Why not do table tests for these?  They should be doable as a table test pretty easily for less lines.

File: pkg/cloudprovider/providers/gce/gce.go [hunk 2]
```diff
@@ -100,6 +100,30 @@ type Config struct {
 	}
 }
 
+type Disks interface {
```
Comment (pmorie) [inline pkg/cloudprovider/providers/gce/gce.go]: Godoc on type?  this is exported.

File: pkg/volume/gce_pd/attacher_test.go [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+	ret        error
+}
+
+type diskIsAttachedCall struct {
+	diskName, instanceID string
+	isAttached           bool
+	ret                  error
+}
+
+type fakeGCE struct {
```
Comment (pmorie) [inline pkg/volume/gce_pd/attacher_test.go]: Nice job w/ this fake

File: pkg/volume/gce_pd/attacher_test.go [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+	attachCalls []attachCall
+	attachIndex int
```
Comment (pmorie) [inline pkg/volume/gce_pd/attacher_test.go]: i find the index name a little confusing, how about attachObservedCalls

**Comment by pmorie** [comment]
LGTM mod a couple nits

**Comment by goltermann** [comment]
@jsafrane can you clean up the code nits soon?  We branch for Beta this Friday.

File: pkg/cloudprovider/providers/gce/gce.go (hunk 2 above)
Comment (jsafrane) [inline pkg/cloudprovider/providers/gce/gce.go]: fixed

**Comment by jsafrane** [comment]
I reworked the unit tests completely, now it uses table approach and uses single expected call instead of array - previously, `AttachDisk/DetachDisk` was called in the loop, now the loop is outside of GCE volume plugin.

**Comment by saad-ali** [comment]
LGTM. Good cleanup

**Pull request merged**